### ISSUE TITLE
Create database, table should failed when names with white space

### ISSUE THREAD

**Issue by jackysp** [pingcap/tidb#3838]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```sql
create table `t1 `(a int); 
create database `db1 `;
create table t1(`a ` int);
```

2. What did you expect to see?

```
mysql> create table `t1 `(a int);
ERROR 1103 (42000): Incorrect table name 't1 '
mysql> create database `db1 `;
ERROR 1102 (42000): Incorrect database name 'db1 '
mysql> create table t1(`a ` int);
ERROR 1166 (42000): Incorrect column name 'a '
```

3. What did you see instead?

```
mysql> create table `t1 `(a int);
Query OK, 0 rows affected (0.01 sec)

mysql> create database `db1 `;
Query OK, 0 rows affected (0.00 sec)

mysql> create table t1(`a ` int);
Query OK, 0 rows affected (0.01 sec)

```

4. What version of TiDB are you using (`tidb-server -V`)?

**Comment by breeswish** [comment]
Clarify for bug fixers, according to doc:

> Database, table, and column names cannot end with space characters.

Notice that leading space or space in elsewhere is allowed.

**Comment by jackysp** [comment]
Fixed #3927
